src/components/SwapForm/index.tsx

```diff
@@ -301,7 +305,10 @@ export default function Swap() {
     }
     // Do evaluate usd select tokens prices
     if (data.size && !isLoading) {
-      data.forEach((token) => {
+      const getAssetList: TokenListWithNotSelectableToken[] = []
+      data.forEach((value) => getAssetList.push(value))
+      const tieNativeToWrapAssetList = tieNativeToWrapToken(getAssetList)
```

Comment:
Why not just, pass data as an arg to tieNativeToWrapToken. It creates a new array and doesn't seem to mutate an incoming one. As for typing you can just assert it if that is the problem.

Reply:
Cause we have already used it at ModalSelectAssets component with plain array data as argument, and moving using map data here as argument will affect that place, and I cut corners here we add them there.

Reply:
I don't get it here - data.forEach((value) => getAssetList.push(value))
you just copy the data array basically
And then calling -  tieNativeToWrapToken
That extra N iterations with same result

Reply:
Because this is JS Map and available sequence here only forEach